src/main/cpp/t07_ip.cpp

```diff
@@ -29,10 +29,41 @@
 
 #include "t07_ip.h"
 #include <iostream>
+#include <string>
 
 
 using namespace std;
 
 int t07_ip() {
+    bool f;
+    f = false;
+    string s, sl;
+    int counter;
+    counter = 0;
+    getline(cin, s);
+    for (int i = 0; i < s.size(); i++) {
+        if (isdigit(s[i]))
+            trio += s[i];
+        else if ((s[i] == '.') && (sl.length()!= 0)) {
+            if (stoi(sl) > 255) {
+                cout << "NO";
+                return 0;
+            }
+            else {
+                counter++;
+                sl = "";
+            }
+        }
+        else {
+            cout << "NO";
+            return 0;
+        };
+    };
 
-}
+    if (counter == 3) f = true;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'f' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4760382)